Template/Tag/MatomoTag.web.js

```diff
@@ -21,9 +21,14 @@
     // We use the stringify and parse to make sure that we have a copy and not a reference
     var localPaq = window._paq && window._paq.length ? JSON.parse(JSON.stringify(window._paq)) : localPaq || [];
     var indexesOfConfigs = [];
+    var hasTrackPageViewViaPaq = false;
     // Clear window._paq to prevent things from being tracked too early
     while (window._paq.length > 0) {
-        window._paq.pop();
+        var poppedItem = window._paq.pop();
+        // Check if trackPageView is set in _paq variable to ensure tracking works even with no callback
+        if (Array.isArray(poppedItem) && poppedItem.toString() == ['trackPageView'].toString()) {
+            hasTrackPageViewViaPaqIndex = true;
```

Comment:
I just realised that the variable names weren't consistent.
```suggestion
            hasTrackPageViewViaPaq = true;
```

Reply:
@snake14 Updated please check